playground/web/index.html

```diff
@@ -0,0 +1,382 @@
+            display: grid;
+            grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
+            gap: 0.5rem;
+            margin-bottom: 1rem;
+        }
+
+        .flag-group {
+            display: flex;
+            align-items: center;
+            gap: 0.5rem;
+        }
+
+        .flag-group input[type="checkbox"] {
+            width: auto;
+        }
+
+        .flag-group label {
+            margin: 0;
+            font-weight: normal;
+            font-size: 0.85rem;
+            cursor: pointer;
+        }
+
+        .output-container {
+            flex: 1;
+            display: flex;
+            flex-direction: column;
+            gap: 1rem;
+        }
+
+        .output-section {
+            border: 1px solid #ddd;
+            border-radius: 4px;
+            overflow: hidden;
+        }
+
+        .output-header {
+            background: #ecf0f1;
+            padding: 0.5rem 0.75rem;
+            font-weight: bold;
+            font-size: 0.85rem;
+            color: #2c3e50;
+            border-bottom: 1px solid #ddd;
+        }
+
+        .output-content {
+            padding: 0.75rem;
+            background: #fafafa;
+            min-height: 60px;
+            max-height: 200px;
+            overflow-y: auto;
+            font-size: 0.85rem;
+            line-height: 1.4;
+        }
+
+        .match-highlight {
+            background: #e74c3c;
+            color: white;
+            padding: 1px 2px;
+            border-radius: 2px;
+            font-weight: bold;
+        }
+
+        .capture-highlight {
+            background: #9b59b6;
+            color: white;
+            padding: 1px 2px;
+            border-radius: 2px;
+            font-weight: bold;
+        }
+
+        .error {
+            color: #e74c3c;
+            background: #fdf2f2;
+            padding: 0.75rem;
+            border-radius: 4px;
+            border: 1px solid #fecaca;
+            font-size: 0.85rem;
+        }
+
+        .success {
+            color: #27ae60;
+            font-weight: bold;
+        }
+
+        .info {
+            color: #7f8c8d;
+            font-style: italic;
+        }
+
+        .toggle-section {
+            margin-bottom: 1rem;
+        }
+
+        .toggle-btn {
+            background: #3498db;
+            color: white;
+            border: none;
+            padding: 0.5rem 1rem;
+            border-radius: 4px;
+            cursor: pointer;
+            font-size: 0.85rem;
+            transition: background-color 0.2s;
+        }
+
+        .toggle-btn:hover {
+            background: #2980b9;
+        }
+
+        .toggle-btn.active {
+            background: #27ae60;
+        }
+
+        .hidden {
+            display: none;
+        }
+
+        .parse-tree {
+            background: #2c3e50;
+            color: #ecf0f1;
+            padding: 1rem;
+            border-radius: 4px;
+            white-space: pre-wrap;
+            font-family: 'Courier New', monospace;
```

Comment:
Not sure why they have a different monospace font here. In general the fonts are bit all over the place currently (e.g. the blue buttons have a different font from the rest). I think we should have:
- Set a generic non-monospace font for all the UI stuff (including the buttons)
- Only use monospace for the input and result text boxes